sympy/functions/elementary/piecewise.py

```diff
@@ -806,6 +807,25 @@ def _eval_nseries(self, x, n, logx, cdir=0):
         args = [(ec.expr._eval_nseries(x, n, logx), ec.cond) for ec in self.args]
         return self.func(*args)
 
+    def _eval_factor(self):
+        from sympy.polys import factor, cancel
+        from sympy.polys.polyerrors import PolynomialError
+
+        args = [(e if hasattr(e, "_eval_factor") else factor(e), c) for e, c in self.args]
+        commonfact = None
+        for expr, cond in args:
+            if commonfact is not None:
+                try:
+                    commonfact = gcd(commonfact, expr)
+                except PolynomialError:
+                    commonfact = 1
```

Comment:
Better to use `S.One` here.

Reply:
Yes, I have nothing against it, but how is it better or worse?

Reply:
1 is a plain Python integer. S.One is a SymPy integer. So 1 will have to be converted to a SymPy integer when used. Here on line 827, so two times per argument. One will probably not notice the speed difference (but there is one), more like good practice.

Reply:
Okay, I replace it. I imagine is the same think for the sets in _eval_rewrite_as_sign ?